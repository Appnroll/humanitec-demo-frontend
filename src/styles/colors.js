export const colors = {
  primaryColor: '#1b51a0',
  primaryColorLighter: '#dce8f9',
  primaryColorLightest: '#ebf2fc',
  textPrimaryColor: '#191919',
  textCaption: 'rgba(102, 102, 102, 1)',
  textSecondaryColor: '#1b51a0',
  backgroundPrimaryColor: '#eceff3',
  backgroundSecondaryColor: '#f9fafb',
  dangerColor: '#DD0000',
  white: '#fff',
  black: '#000',
  grayDark: '#222',
  gray: '#333333',
  grayMediumDark: '#706e6b',
  grayMedium: '#8B8D9B',
  grayLight: '#9B9B9B',
  grayLighter: '#dfdfdf',
  grayVeryLight: '#F5F5F5',
  grayVeryLighter: '#ececec',
}

