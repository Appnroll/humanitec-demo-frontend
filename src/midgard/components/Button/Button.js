import styled from 'styled-components';
import { colors } from 'styles/colors';

export const Button = styled.button`
  background-color: ${colors.primaryColor};
  border: none;
  border-radius: rem-calc(4);
  height: rem-calc(32);
  padding: rem-calc(0 12);
  transition: all 0.2s linear;
  font-weight: 600;
  font-size: rem-calc(12);
  color: ${colors.white};
  cursor: pointer;
  outline: none;

  &:hover {
    background-color: darken($primary-color, 5);
  }

  &:disabled {
    background-color: $gray-light;
    color: ${colors.gray};
    cursor: default;
  }
`;
