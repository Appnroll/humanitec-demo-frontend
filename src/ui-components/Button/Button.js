import styled, { css } from 'styled-components'
import { colors } from 'colors'
import { rem } from 'polished'

export const Button = styled.button`
  display: inline-block;
  border-radius: ${rem(4)};
  height: ${rem(40)};
  padding: 0 ${rem(12)};
  transition: all 0.2s linear;
  font-weight: 600;
  background-color: ${colors.primary};
  border: ${rem(1)} solid ${colors.primary};
  color: ${colors.white};
  cursor: pointer;
  outline: none;
  margin: ${rem(4)} 0;

  &:hover {
    background-color: ${colors.primaryDarker};
  }

  &:disabled {
    background-color: ${colors.grayLight};
    border-color: ${colors.grayLight};
    color: ${colors.gray};
    cursor: default;
  }

  ${props => props.secondary && css`
    background: ${colors.white};
    color: ${colors.primary};
    border-color: ${colors.primary};

    &:hover {
      background-color: ${colors.primaryLightest};
    }
  `}

  ${props => props.small && css`
    height: ${rem(32)};
  `}
`