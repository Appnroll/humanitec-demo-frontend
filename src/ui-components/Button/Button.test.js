import React from 'react'
import renderer from 'react-test-renderer'
import { Button }  from 'ui-components/Button/Button.js'
import { colors } from 'colors'
import 'jest-styled-components'

test('simple button test', () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('color', colors.white)
    expect(tree).toHaveStyleRule('background-color', colors.primary)
})

test('secondary button test', () => {
    const tree = renderer.create(<Button secondary />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('color', colors.primary)
    expect(tree).toHaveStyleRule('background-color', colors.primary)
})

test('upload button test', () => {
    const tree = renderer.create(<Button small upload />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('position', 'relative')
})