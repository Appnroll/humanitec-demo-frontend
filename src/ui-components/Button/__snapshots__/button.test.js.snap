// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`secondary button test 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  height: 2.5rem;
  padding: 0 0.75rem;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  font-weight: 600;
  background-color: #1B51A0;
  border: 0.0625rem solid #1B51A0;
  color: #FFFFFF;
  cursor: pointer;
  outline: none;
  margin: 0.25rem 0;
  background: #FFFFFF;
  color: #1B51A0;
  border-color: #1B51A0;
}

.c0 img {
  max-height: 1.125rem;
  max-width: 1.125rem;
  margin-right: 0.25rem;
}

.c0:hover {
  background-color: #143b74;
}

.c0:disabled {
  background-color: #b4b4b4;
  border-color: #b4b4b4;
  color: #686868;
  cursor: default;
}

.c0:hover {
  background-color: #c5d8f5;
}

<button
  className="c0"
/>
`;

exports[`simple button test 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  height: 2.5rem;
  padding: 0 0.75rem;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  font-weight: 600;
  background-color: #1B51A0;
  border: 0.0625rem solid #1B51A0;
  color: #FFFFFF;
  cursor: pointer;
  outline: none;
  margin: 0.25rem 0;
}

.c0 img {
  max-height: 1.125rem;
  max-width: 1.125rem;
  margin-right: 0.25rem;
}

.c0:hover {
  background-color: #143b74;
}

.c0:disabled {
  background-color: #b4b4b4;
  border-color: #b4b4b4;
  color: #686868;
  cursor: default;
}

<button
  className="c0"
/>
`;

exports[`upload button test 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  height: 2.5rem;
  padding: 0 0.75rem;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  font-weight: 600;
  background-color: #1B51A0;
  border: 0.0625rem solid #1B51A0;
  color: #FFFFFF;
  cursor: pointer;
  outline: none;
  margin: 0.25rem 0;
  height: 2rem;
  position: relative;
}

.c0 img {
  max-height: 1.125rem;
  max-width: 1.125rem;
  margin-right: 0.25rem;
}

.c0:hover {
  background-color: #143b74;
}

.c0:disabled {
  background-color: #b4b4b4;
  border-color: #b4b4b4;
  color: #686868;
  cursor: default;
}

.c0 input {
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  opacity: 0;
}

<button
  className="c0"
/>
`;
